language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/LSPackageLoader.java

```diff
@@ -175,6 +186,22 @@ private List<PackageInfo> getPackagesFromRepository(PackageRepository repository
         return packages;
     }
 
+    public List<PackageInfo> updatePackageMap(DocumentServiceContext context) {
+        Optional<Project> project = context.workspace().project(context.filePath());
```

Comment:
Can't we write unit tests without mocking this method?

Reply:
We would have to pull packages from central in that case. When we pull, the .ballerina folder of the local computer will be modified and we would have to read from it. In test env, if we read from the remote repository of the .ballerina, that can lead to inconsistencies. Therefore, it was decided to mock the update method.

Reply:
I see. No issue then. Let's merge this.